### PR TITLE
blocks: use VOLK popcnt implementations for count_bits

### PR DESCRIPTION
**Pull request by ncorgan**

### CODE REVIEW

**Comment by willcode** [comment]
@ncorgan need a signoff for DCO.

File: gr-blocks/lib/count_bits.cc [hunk 1]
```diff
@@ -45,16 +41,18 @@ unsigned int count_bits16(unsigned int x)
...
+    return res;
 }
 
 unsigned int count_bits64(unsigned long long x)
 {
-    return count_bits32((x >> 32) & 0xffffffff) + count_bits32(x & 0xffffffff);
+    uint64_t res_as_u64 = 0;
+    volk_64u_popcnt(&res_as_u64, static_cast<uint64_t>(x));
+
+    return static_cast<unsigned long long>(res_as_u64);
```
Comment (marcusmueller) [inline gr-blocks/lib/count_bits.cc:55]: hm, but you're return an `unsigned int`, not an `unsigned long long`; and frankly, that should suffice for the near future :)

**Review by marcusmueller — CHANGES_REQUESTED** [review]

File: gr-blocks/lib/count_bits.cc (hunk 1 above)
Comment (mbr0wn) [reply gr-blocks/lib/count_bits.cc:55]: While we're touching this, can you please change this to uint64_t?

Comment (mbr0wn) [reply gr-blocks/lib/count_bits.cc:55]: Sorry I meant the function argument, not the return value.

Comment (ncorgan) [reply gr-blocks/lib/count_bits.cc:55]: What, where did that odd mistake come from...I'll make these changes when I get home from work.

Comment (ncorgan) [reply gr-blocks/lib/count_bits.cc:55]: I'd argue the return type is also wrong. Should `count_bits64` be returning a 32-bit number?

Comment (ncorgan) [reply gr-blocks/lib/count_bits.cc:55]: Derp, ignore my uint64_t comment, I was thinking byteswap for some reason.

**Review by mbr0wn — APPROVED** [review]

**Pull request merged**